packages/babel-generator/src/printer.js

```diff
@@ -231,7 +231,7 @@ export default class Printer extends Buffer {
     if (this.format.shouldPrintComment) {
       return this.format.shouldPrintComment(comment.value);
     } else {
-      if (comment.value.indexOf("@license") >= 0 || comment.value.indexOf("@preserve") >= 0) {
+      if (!this.format.compact && comment.value.indexOf("@license") >= 0 || comment.value.indexOf("@preserve") >= 0) {
```

Comment:
The precedence works out. || has higher precedence

On Monday, December 7, 2015, René Kooi <notifications@github.com> wrote:

> In packages/babel-generator/src/printer.js
> <https://github.com/babel/babel/pull/3146#discussion_r46809012>:
>
> > @@ -231,7 +231,7 @@ export default class Printer extends Buffer {
> >      if (this.format.shouldPrintComment) {
> >        return this.format.shouldPrintComment(comment.value);
> >      } else {
> > -      if (comment.value.indexOf("@license") >= 0 || comment.value.indexOf("@preserve") >= 0) {
> > +      if (!this.format.compact && comment.value.indexOf("@license") >= 0 || comment.value.indexOf("@preserve") >= 0) {
>
> should this be !compact && (@license || @preserve)?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/babel/babel/pull/3146/files#r46809012>.
>